cms-contentful/it/com/commercetools/sunrise/cms/contentful/ContentfulCmsServiceIT.java

```diff
@@ -52,65 +46,221 @@ public void whenCouldNotFetchEntry_thenThrowException() throws Exception {
-        Optional<CmsPage> content = waitAndGet(contentfulCmsService.page("finn", Collections.singletonList(Locale.ITALIAN)));
+    public void whenAskForExistingStringContentWithNotDefinedLocale_thenReturnEmpty() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), PAGE_TYPE_NAME, PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = waitAndGet(contentfulCmsService.page("finn", singletonList(Locale.ITALIAN)));
+
         assertThat(content).isEmpty();
     }
 
     @Test
-    public void whenAskForNotExistingStringContentThenNotPresent() throws Exception {
+    public void whenAskForNotExistingStringContent_thenReturnEmpty() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), PAGE_TYPE_NAME, PAGE_TYPE_ID_FIELD_NAME);
         Optional<CmsPage> content = waitAndGet(contentfulCmsService.page("finn", SUPPORTED_LOCALES));
         assertThat(content).isPresent();
         assertThat(content.get().field("pageContent.notExistingField")).isEmpty();
     }
 
     @Test
-    public void whenAskForExistingAssetContentThenGet() throws Exception {
+    public void whenAskForExistingAssetContent_thenGet() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), PAGE_TYPE_NAME, PAGE_TYPE_ID_FIELD_NAME);
+
         Optional<CmsPage> content = waitAndGet(contentfulCmsService.page("jacke", SUPPORTED_LOCALES));
+
         assertThat(content).isPresent();
         assertThat(content.get().fieldOrEmpty("pageContent.image")).isEqualToIgnoringCase("//images.contentful.com/l6chdlzlf8jn/2iVeCh1FGoy00Oq8WEI2aI/93c3f0841fcf59743f57e238f6ed67aa/jake.png");
     }
 
     @Test
-    public void whenAskForNotExistingAssetContentThenNotPresent() throws Exception {
+    public void whenAskForNotExistingAssetContent_thenReturnEmpty() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), PAGE_TYPE_NAME, PAGE_TYPE_ID_FIELD_NAME);
+
         Optional<CmsPage> content = waitAndGet(contentfulCmsService.page("jacke", SUPPORTED_LOCALES));
+
         assertThat(content).isPresent();
         assertThat(content.get().fieldOrEmpty("pageContent.notExistingAsset")).isEmpty();
 
     }
 
+    @Test
+    public void whenAskForContentInArray_thenGetElement() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("array[0].name");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("author1");
```

Comment:
you're right. I changed it because I was misled by the name of the method .contains which in String class has different semantics.
So I will change it to AssertJ's .hasValue() which is an alias for .contains() and imo is much more adequate to its behavior.